### PR TITLE
fix: render popover template content

### LINKED ISSUE #1252: [notebooks-v1] Date-time popover renders empty content and accumulates overlays
## Summary

The shared Angular `DateTimeComponent` can create a visible popover card with no Local or UTC content. Repeated hovering can leave additional empty overlay cards in the document.

This behavior affects consumers of the Kubeflow common library pinned at revision `0b835a7848ddcadf2c920a8e14aeae121ac982d9`. The component remains present on the `notebooks-v1` branch.

## Reproduction

1. Render a resource table column with `DateTimeValue`.
2. Provide a valid creation timestamp.
3. Hover over the relative-time text rendered by `lib-date-time > .truncate`.
4. Move the pointer away and repeat the hover.

## Expected behavior

The popover displays the Local and UTC timestamp values, and repeated hovering does not accumulate overlay cards.

## Actual behavior

The overlay contains an empty card:

```html
<lib-popover>
  <mat-card class="popover-card"><!-- empty projected content --></mat-card>
</lib-popover>
```

Repeated hovering can create additional blank cards.

The date-time template binds `[libPopover]="timeTpl"` directly to the inner `.truncate` element. The issue appears to involve template portal projection or change-detection propagation when the overlay is created.

## Downstream evidence

KServe models-web-app pull request #196 reproduces the problem in a production image and applies a narrowly scoped downstream workaround:

https://github.com/kserve/models-web-app/pull/196

The workaround disables pointer-event capture only for the affected Endpoint creation-time trigger while retaining `DateTimeValue` sorting, filtering, relative-time rendering, and empty-date behavior.

## Suggested investigation

Please inspect the popover template portal and OnPush change-detection path used by `DateTimeComponent`. The downstream workaround can be removed when the shared component either renders the template reliably or exposes a supported input for disabling its popover.

### PR DESCRIPTION
**Pull request by danish9039**
## Problem

`PopoverDirective` creates and attaches the overlay component before assigning its message or template portal. A deferred `markForCheck()` does not render newly assigned content before the dynamically attached card becomes visible, so the date-time popover can appear empty.

## Change

- Perform an immediate overlay view check in `show()` after assigning the message, template portal, and classes.
- Keep `updatePosition()` on deferred `markForCheck()` so dynamic string-input updates do not trigger synchronous change detection during the parent check.
- Register one `afterHidden()` subscription with `take(1)` per attached popover instance so rapid re-entry cannot accumulate duplicate detach callbacks.
- Add regression coverage for repeated date-time template hovers, rendering and updating a string-message popover while shown, and rapid re-entry before a pending hide completes.
- Verify that leaving removes the overlay pane.

## Red-green evidence

With the original behavior, the date-time regression test fails because the overlay text is empty. With the immediate view check in `show()`, Local and UTC values render across repeated hover cycles, while dynamic string messages also render and update correctly. The rapid re-entry test fails with two `detach()` calls before the lifecycle correction and passes with one subscription per popover instance.

## Verification

- `npm run format:check`
- `npm run lint-check`
- `npm run test:prod` (63 tests passed)
- `npm run build`

## Fixes
 #1252

----
Assisted-by: OpenAI Codex (GPT-5)

### CODE REVIEW

**Comment by danish9039** [comment]
@juliusvonkohout @christian-heusel @thesuperzapper

**Comment by danish9039** [comment]
Before, we were having empty white cards when hovering on links, look after 39 seconds 

https://github.com/user-attachments/assets/a66fb160-6eab-405b-967d-bb3a9e8610d0


After fix 



 [Screencast from 2026-07-16 18-34-50.webm](https://github.com/user-attachments/assets/c284c6cc-1865-4821-a824-ce11f751462e)

File: components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/popover/popover.directive.ts [hunk 1]
```diff
@@ -107,6 +107,7 @@ export class PopoverDirective implements OnDestroy {
     if (this.libPopoverClass.length > 0) {
       this.popoverInstance.classList = this.libPopoverClass;
     }
+    this.popoverInstance.detectChanges();
     this.popoverInstance.show(delay);
     this.popoverInstance.afterHidden().subscribe(() => this.detach());
```
[the remark this answers is not in the document]
Comment (danish9039) [reply components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/popover/popover.directive.ts]: Moved the one-shot `afterHidden()` subscription into `createPopover()` and added a rapid re-entry regression test that verifies each popover instance detaches exactly once.

**Comment by christian-heusel** [comment]
/cc @paulovmr @thaorell
for the needed frontend knowledge

**Review by christian-heusel — COMMENTED** [review]
Thanks a lot for your contribution! 🤗 

Did you use AI in any capacity while creating these changes? I see that your commits do not carry any `Assisted-By: <...>` / `Co-authored-by: <...>` tags, so I wanted to make sure that you're aware of the [Kubeflow AI Policy](https://www.kubeflow.org/docs/about/ai_policy/). 🤗 🤖

**Comment by LogicalGuy77** [comment]
hi @christian-heusel , @danish9039 has added AI usage in the PR description, can we move forward with merging this PR?

**Comment by juliusvonkohout** [comment]
Is this still needed given https://github.com/kserve/models-web-app/commit/2d8ec7d8aa18133aa9e481fdd996b93559491077 and https://github.com/kserve/models-web-app/commit/2d8ec7d8aa18133aa9e481fdd996b93559491077 ? Please verify with the lates images

**Comment by LogicalGuy77** [comment]
hi @juliusvonkohout I have tested against the latest models-web-application image and the root cause has been fixed. We can safely close this PR and #1252 

we can also close https://github.com/kserve/models-web-app/pull/196
